primer/brand
# Support n color themes in StyleDictionary custom formatter

## Summary

The style dictionary custom formatter for generating color tokens has been refactored to support additional color modes beyond `light` and `dark`. This is in preparation to upstream the formatter to Primer Primitives. 

## List of notable changes:




- **updated** `color-mode-attributes.js` file to handle new color modes
- **removed** duplication of light and root by comma separating the selectors (see screenshot below) 
- **added** new test coverage to ensure there are no regressions going forward and cover edge cases


## Steps to test:



1. Check the CI diff comment below and ensure there are no design token changes 
1. Check the new unit tests pass

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

There is now an `allowList` of supported modes. 
![Screenshot 2022-09-13 at 11 30 22](https://user-images.githubusercontent.com/13340707/189879145-868e7b2d-13c1-436c-880c-05581ee2027b.png)

**Primer Brand only requires 'dark' and 'light', but as we don't provide values for these in our property files, they won't get built.**

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

```json
{
  "brand": {
    "InlineLink": {
      "color": {
        "rest": {
          "value": "var(--base-color-scale-blue-5)",
          "dark": "var(--base-color-scale-blue-4)",
          "dark_dimmed": "var(--base-color-scale-blue-6)"
        },
        "pressed": {
          "value": "var(--base-color-scale-blue-7)",
          "dark": "var(--base-color-scale-blue-2)",
          "dark_dimmed": "var(--base-color-scale-blue-6)"
        }
      }
    }
  }
}
```

 </td>
<td valign="top">

```css
/**
 * Do not edit directly
 * Generated on Tue, 13 Sep 2022 10:33:30 GMT
 */

:root,
[data-color-mode="light"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-5);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-7);
}

[data-color-mode="dark"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-4);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-2);
}

[data-color-mode="dark_dimmed"] {
  --brand-InlineLink-color-rest: var(--base-color-scale-blue-6);
  --brand-InlineLink-color-pressed: var(--base-color-scale-blue-6);
}
```

</td>
</tr>
</table>
